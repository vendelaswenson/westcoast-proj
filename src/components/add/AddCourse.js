function AddCourse() {
    return (
        <>
        <h2>Add course</h2>
      <form>
        <label>Course name:
          <input type="text" />
        </label>
        <label>Course number:
          <input type="number" />
        </label>
        <button>Add</button>
      </form>
      </>
    )
  }

  export default AddCourse