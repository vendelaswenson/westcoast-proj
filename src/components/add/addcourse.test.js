import AddCourse from './AddCourse';
import {render, cleanup, screen} from '@testing-library/react'
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';



afterEach(() => {
    cleanup(); 
})

describe("Component", () => {
  it("renders without crashing", async () => {
    const utils = render(<AddCourse />, {wrapper: MemoryRouter});
    expect(utils).toMatchSnapshot();
    expect(utils).toBeTruthy();
  });

  it("Length label should be in the document", () => {
    render(<AddCourse />, {wrapper: MemoryRouter});
    const inputNode = screen.getByLabelText('Length:')
    expect(inputNode).toBeInTheDocument();
  })
});
