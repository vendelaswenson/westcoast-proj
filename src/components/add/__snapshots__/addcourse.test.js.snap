// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="course-container"
      >
         
        <h2
          class="add-h2"
        >
          Add Course
        </h2>
        <form>
          <div>
            <label
              for="name"
            >
              Course name:  
            </label>
            <input
              id="name"
              name="name"
              type="text"
            />
          </div>
          <div>
            <label
              for="number"
            >
              Course number:  
            </label>
            <input
              id="number"
              name="number"
              type="text"
            />
          </div>
          <div>
            <label
              for="length"
            >
              Length:  
            </label>
            <input
              id="length"
              name="length"
              type="text"
            />
          </div>
          <div>
            <label
              for="description"
            >
              Description:  
            </label>
            <input
              id="description"
              name="description"
              type="text"
            />
          </div>
          <div>
            <label
              for="startDate"
            >
              Start date:  
            </label>
            <input
              id="startDate"
              name="startDate"
              type="text"
            />
          </div>
          <div
            class="btns-container"
          >
            <button
              class="btn"
            >
              Add
            </button>
            <button
              class="btn"
            >
              Cancel
            </button>
          </div>
        </form>
         
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="course-container"
    >
       
      <h2
        class="add-h2"
      >
        Add Course
      </h2>
      <form>
        <div>
          <label
            for="name"
          >
            Course name:  
          </label>
          <input
            id="name"
            name="name"
            type="text"
          />
        </div>
        <div>
          <label
            for="number"
          >
            Course number:  
          </label>
          <input
            id="number"
            name="number"
            type="text"
          />
        </div>
        <div>
          <label
            for="length"
          >
            Length:  
          </label>
          <input
            id="length"
            name="length"
            type="text"
          />
        </div>
        <div>
          <label
            for="description"
          >
            Description:  
          </label>
          <input
            id="description"
            name="description"
            type="text"
          />
        </div>
        <div>
          <label
            for="startDate"
          >
            Start date:  
          </label>
          <input
            id="startDate"
            name="startDate"
            type="text"
          />
        </div>
        <div
          class="btns-container"
        >
          <button
            class="btn"
          >
            Add
          </button>
          <button
            class="btn"
          >
            Cancel
          </button>
        </div>
      </form>
       
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
